### PR TITLE
Allow `Checkbox`es in `DataTable`s to inherit colors from `CheckboxTheme`

### LINKED ISSUE #96004: `Checkbox`es in `DataTable`s should inherit colors from `CheckboxTheme`
## Proposal

`Checkbox`es in`DataTable`s currently are still hardcoded to have `checkColor` and `fillColor` set by `colorScheme.primary` and 
`colorScheme.onPrimary`. If we just remove those values, they will fall back to how they're configured internally by the `CheckboxTheme`. This isn't my idea—there's a stale todo in the DataTable implementation noting that we should remove those values when #53420 is closed (which it is):

image

Here's the way it looks now:

### PR DESCRIPTION
**Pull request by willlockwood**
This PR fixes https://github.com/flutter/flutter/issues/96004

This PR resolves a stale TODO left in the `DataTable` code (see the issue for details), by removing the hardcoded values from `DataTable` controlling the checkboxes colors, which instead allows them to inherit those values for the `checkColor` and `fillColor` from the `CheckboxTheme` rather than `ColorScheme.primary` and `ColorScheme.onPrimary`. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/material/data_table_test.dart
Comment (werainkhatri) [inline packages/flutter/test/material/data_table_test.dart:1353]: IMO it would be cleaner if this is tested the same way it's done in `checkbox_theme_test.dart`. WDYT?

https://github.com/flutter/flutter/blob/d719463489218f9de5f7b8b2237cb256f7617e9f/packages/flutter/test/material/checkbox_theme_test.dart#L138-L142

```suggestion
    expect(
        Material.of(tester.element(find.byType(Checkbox).last)),
        paints
          ..path()
          ..path(color: fillColor)
          ..path(color: checkColor)
    );
```

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/data_table_test.dart
Comment (willlockwood) [reply packages/flutter/test/material/data_table_test.dart:1353]: Sure! Will do momentarily. Thanks for the review 👍🏻

**Review by werainkhatri — APPROVED** [review]
LGTM. Thank you for your contribution.

Although, this would need another reviewer's approval to land, as per our new commit policy.

**Comment by willlockwood** [comment]
> Although, this would need another reviewer's approval to land, as per our new commit policy.

Great, sounds good, thanks for the review! Any suggestions for how I should go about trying to get another pair of eyes on it? If the best practice is to just wait until another reviewer finds it among the list of PRs, that's totally fine with me, just not exactly sure what the etiquette/expectation is for how that usually works 👍🏻

**Comment by werainkhatri** [comment]
PRs should get reviewed within about a week when the relevant team's PR review meeting happens (might be a bit longer with the holiday in the US right now).

File: packages/flutter/lib/src/material/data_table.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/data_table.dart:704]: Nice thanks for the clean up! :)

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
 
 Thank you for the contribution!

**Pull request merged**
